language/move-prover/boogie-backend/src/spec_translator.rs

```diff
@@ -729,24 +763,31 @@ impl<'env> SpecTranslator<'env> {
     ) {
         let loc = self.env.get_node_loc(node_id);
         // Translate range expressions.
-        let mut range_tmps = HashMap::new();
+        let mut vec_range_tmps = HashMap::new();
```

Comment:
Wondering whether it is time to break this  function down by introducing a few helper functions, or specializing for the multiple cases.

Reply:
TBH I haven't found a non-artificial way to break down this function.

Reply:
SG. It was more a generic review comment.